### PR TITLE
Improve text when there is no data available

### PR DESCRIPTION
**Pull request by clovis1122**
If the selected application or range does not have data to show, we'll render a message saying so. Additionally, we won't show the "Frame width represents" text if there's nothing to show.

closes https://github.com/pyroscope-io/pyroscope/issues/483

image

### CODE REVIEW

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx [hunk 1]
```diff
@@ -139,7 +142,18 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
...
+        {dataUnavailable ? (
+          <div className="error-message">
+            <span>
+              No profiling data available for this application / time range.
+            </span>
+          </div>
+        ) : null}
+        <div
+          style={{
+            opacity: dataUnavailable ? 0 : 1,
```
Comment (eh-am) [inline webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx:154]: I'm a bit curious, why was this needed, as opposed to simply not rendering the canvas?

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx [hunk 2]
```diff
@@ -139,7 +142,18 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
           ExportData={ExportData}
         />
 
-        <div>
+        {dataUnavailable ? (
+          <div className="error-message">
```
Comment (eh-am) [inline webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx:146]: If this class is used only here, would you mind using css modules? We are now trying to scope the css as much as possible.

Just to give you some direction, you have to create a `index.css.module`, for example
https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/FlameGraph/FlameGraphComponent/Highlight.tsx#L2
https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/FlameGraph/FlameGraphComponent/Highlight.module.css

**Review by eh-am — COMMENTED** [review]
Awesome, thank you for your contribution!

Would you mind adding a unit test for this case? I'm thinking something like passing an empty`flamebearer` and assert that the "No profiling data" message is shown and the flamegraph is not visible.

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx (hunk 1 above)
Comment (clovis1122) [reply webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx:154]: Because of the logic attached to `canvasRef`. I imagine simply not rendering the canvas is possible, though it would require some larger refactor across different components.

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx (hunk 2 above)
Comment (clovis1122) [reply webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx]: Done!

**Comment by clovis1122** [comment]
hmmm, I added some tests but had to move it to the bottom... it looks like some of the tests in this file have dependency among themselves.

**Comment by eh-am** [comment]
Nice!

> hmmm, I added some tests but had to move it to the bottom... it looks like some of the tests in this file have dependency among themselves.

Almost! The [`findBy` family returns a promise](https://testing-library.com/docs/queries/about/) and the test isn't waiting for it. I suggest to just replace by `getBy`. Plus, you can use a regex there, to avoid matching the whooooole text, eg. `screen.getByText(/No profiling data available/);`


Another thing, can you fix the test that's breaking? We've updated the `FlamegraphComponent` API, it should be pretty straightforward to fix it, `{ i: -1, j: -1}` becomes `Option.none()`

**Comment by clovis1122** [comment]
@eh-am done, this should be good for a second pass!
